### DISCUSSION TITLE
How to initialise ModelSchema instances explicitly?

### DISCUSSION THREAD

**Discussion by oliwarner** [vitalik/django-ninja#880] [Q&A] [answered]
I'm making a TV listing API and I've written an endpoint that should return a dict of dates mapping to a list of episodes for that day. I haven't defined the outer schema, but the model Episode has a EpisodeSchema defined which does a few things.

```
@api.get("/schedule/")
def schedule(request):
    # ... my logic to group episodes into dates ...
    # then return
    return {
        d: [
            EpisodeSchema(ep)
            for ep in grouped_episodes[d]
        ]
        for d in dates
    }
```

But if I do that, it explodes saying (on the EpisodeSchema constructor): `TypeError: __init__() takes exactly 1 positional argument (2 given)`. I've tried a few other incantations instance=ep, obj=ep, etc and give me a `pydantic.error_wrappers.ValidationError` complaining that I haven't set the schema fields.

Is there a way to build a ModelSchema instance from a model instance? Or am I going about this the wrong way?

**Comment by vitalik** [comment, marked as answer]
@oliwarner 

try like this:

```Python
@api.get("/schedule/")
def schedule(request):
    # ... my logic to group episodes into dates ...
    # then return
    return {
        d: [
            EpisodeSchema.model_validate(ep)
            for ep in grouped_episodes[d]
        ]
        for d in dates
    }
```

or with response definition:

```Python
@api.get("/schedule/", response=dict[str, list[EpisodeSchema])
def schedule(request):
    # ... my logic to group episodes into dates ...
    # then return
    return {
        d: list(grouped_episodes[d]) for d in dates
    }
```

**Reply by oliwarner** [reply]
These both work. Thank you so much.
